go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -358,3 +358,60 @@ func updateCondition(conditions []metav1.Condition, cond *metav1.Condition) ([]m
 	}
 	return conditions, false
 }
+
+// UpdateSubsystemError may be used by other controllers handling UDN/NAD/network setup to report any problems that
+// may affect UDN functionality.
+// FieldManager should be unique for every subsystem. When errorMessage is empty, condition will be removed.
+// If given network is not managed by a UDN, no error will be returned.
+func (c *Controller) UpdateSubsystemError(networkName string, errorMessage string, fieldManager string) error {
+	// try to find udn using network name
+	udnNamespace, udnName := template.ParseNetworkName(networkName)
+	if udnName == "" {
+		return nil
+	}
+	udn, err := c.udnClient.K8sV1().UserDefinedNetworks(udnNamespace).Get(context.Background(), udnName, metav1.GetOptions{})
+	if err != nil {
+		return nil
+	}
+	udnStatus := udnapplyconfkv1.UserDefinedNetworkStatus()
+	conditionType := fmt.Sprintf("SubsystemError.%s", fieldManager)
+	if errorMessage != "" {
+		condition := metav1.Condition{
+			Type:               conditionType,
+			Status:             metav1.ConditionTrue,
+			Reason:             "SubsystemError",
+			Message:            errorMessage,
+			LastTransitionTime: metav1.Now(),
+		}
+		udnStatus = udnStatus.WithConditions(condition)
+		for _, statusCondition := range udn.Status.Conditions {
+			if statusCondition.Type == condition.Type && statusCondition.Message == condition.Message {
```

Comment:
I think the comparison `statusCondition.Message == condition.Message` should be a concern of the caller.

Reply:
I still think we shouldn't be comparing because it is racey.

But don't you want to record the events even if condition didn't change?

Reply:
You could use `FindStatusCondition` here as well